scripts/plot_timeline.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+A script for plotting event log files.
+
+Sample use:
+    plot_timeline.py \
+        --dataset A /my/path/A/events.log \
+        --dataset B /my/path/B/events.log \
+        -o timeline.png
+
+To see full command line options documentation, run:
+  plot_timeline.py --help
+"""
+
+import argparse
+import csv
+import matplotlib
+matplotlib.use('Agg') # Allows to work without X dispaly
+import matplotlib.pyplot as plt
+import numpy as np
+import os.path
+import sys
+
+nanos_per_second = 10**9
+nanos_per_milli = 10**6
+
+def get_frames_per_second(frame_present_events):
+    first_second = frame_present_events[0][0] // nanos_per_second
+    last_second = int(np.ceil(frame_present_events[-1][0] / nanos_per_second))
+    duration_seconds = last_second - first_second + 1
```

Comment:
I suggest leaving a comment saying this function buckets frames by second and what the return values are (at first glance I wondered why whole numbers are used, but it makes sense when bucketing by second). Would be good to do the same for some of the other functions as well

Reply:
Thanks for the suggestion, done. The part of the code that calculates FPS is almost the same as in `analyze_frametimes.py`, so it might be worth factoring this out into a common library file in the future, should the code continue to grow.